src/savegame.h

```diff
@@ -0,0 +1,229 @@
+    uint8_t aquatic;
+    uint8_t subterranean;
+    uint8_t largeHomeworld;
+    int8_t richHomeworld;
+    uint8_t artifactsHomeworld;
+    uint8_t cybernetic;
+    uint8_t lithovore;
+    uint8_t repulsive;
+    uint8_t charismatic;
+    uint8_t uncreative;
+    uint8_t creative;
+    uint8_t tolerant;
+    uint8_t fantasticTraders;
+    uint8_t telepathic;
+    uint8_t lucky;
+    uint8_t omniscience;
+    uint8_t stealthyShips;
+    uint8_t transDimensional;
+    uint8_t warlord;
+    uint8_t poorHomeworld;
+};
+
+struct Player {
+    char name[PLAYER_NAME_SIZE];
+    char race[PLAYER_RACE_SIZE];
+    uint8_t picture;
+    uint8_t color;
+    uint8_t personality;
+    uint8_t objective;
+    uint8_t taxRate;
+    uint64_t BC;
+    uint16_t totalFreighters;
+    uint16_t usedFreighters;
+    uint16_t commandPoints;
+    int16_t totalProduction;
+    uint16_t RP;
+    uint16_t food;
+    uint16_t yearlyBC;
+    uint16_t researchProgress;
+    /* Research Area
+     *
+     * 0x00 = None
+     * 0x04 = Construction
+     * 0x07 = Force Fields
+     * 0x0a = Sociology
+     * 0x12 = Biology
+     * 0x16 = Chemistry
+     * 0x1c = Computers
+     * 0x37 = Power
+     * 0x39 = Physics
+     */
+    uint16_t researchArea;
+    uint16_t researchItem;
+    RacePicks racePicks;
+};
+
+struct Star {
+    char name[STARS_NAME_SIZE];
+    uint16_t x;
+    uint16_t y;
+    /*
+    * 0=Large
+    * 1=Medium
+    * 2=Small
+    * 3=Tiny
+    */
+    uint8_t size;
+    /*
+    * 0-7: player id
+    * 0xFF: no owner
+    */
+    uint8_t owner;
+    uint8_t pictureType;
+    /*
+    * 0=Blue
+    * 1=White
+    * 2=Yellow
+    * 3=Orange
+    * 4=Red
+    * 5=Brown
+    * 6=Black hole
+    */
+    uint8_t spectralClass;
+    // Remembers the last selected planet for the system for each player
+    uint8_t lastPlanetSelected[MAX_PLAYERS];
+    // Precomputed bitfield that tells wether there is a black hole between two stars
+    uint8_t blackHoleBlocks[(MAX_STARS + 7) / 8];
+    /*
+    * 0=none
+    * 2=Space Debris
+    * 3=Pirate Cache
+    * 4=Gold Deposites
+    * 5=Gem Deposits
+    * 6=Natives
+    * 7=Splinter Colony
+    * 8=Hero
+    * A=Artifacts world
+    * B=Orion
+    */
+    uint8_t special;
+    // 0-7F = system, FF = no wormhole
+    uint8_t wormhole;
+    // Bitmask that tells if the system is visited by which player
+    uint8_t blockaded;
+    // Bitmask as booleans for each player
+    uint8_t visited;
+    // 0 = none, 1-8 = owner player 0-7
+    uint8_t indictor;
+    // 0 = none, 1-8=owner player 0-7
+    uint8_t artemis;
+};
+
+struct GameState  {
+    struct GameConfig _gameConfig;
+    struct Galaxy _galaxy;
+    uint8_t _starCount;
+    Star _stars[MAX_STARS];
+    struct Heroes _heroes[HERO_COUNT];
+    struct Player _players[PLAYER_COUNT];
+};
+
+class Savegame {
+private:
+    File _file;
+
+public:
+    Savegame(const char *fileName);
+    ~Savegame();
+    GameState* load();
+    void readConfig(File& file, GameState* state);
```

Comment:
It's better to use `SeekableReadStream&` instead of `File&` in function parameters. Then you can for example load the whole file into memory and read data much faster through `MemoryReadStream`.